src/redux/features/categories/types.ts

```diff
@@ -0,0 +1,17 @@
+export interface SubCategory {
+  name: string;
+  _id: string;
+  parentCategoryId: string;
+}
+
+export interface Category {
+  name: string;
+  subCategories: SubCategory[];
+  _v?: number;
+  _id: string;
+}
+
+export interface CategoryState {
+  categories: Category[] | null;
```

Comment:
I recommend using just type array here. It can be empty or filled. It is annoying to manage null and []